### ISSUE TITLE
Formatter should be configurable for a space before and after curly braces in interpolated string

### ISSUE THREAD

**Issue by jamiehankins** [dotnet/roslyn#58015]
**Brief description:**

In most cases, the IDE determines that there should be a space before and after the curly braces in C# code. It does not determine either way for interpolated strings.

We should have a config that determines how the IDE handles this, and the IDE should actively reformat based on this setting.

Suggested name: csharp_space_around_string_insert
Values: false, before, after, before_and_after

**Languages applicable:**

C#

**Code example that the analyzer should report:**

For example, if you create an array and add an initialization list, the IDE adds the spaces for you, like this:
```C#
var values = new int[] { 1, 2, 3 };
```
If you intentionally omit the spaces, then do a "Format Document" command, the IDE adds them back.

However, this is treated differently for an interpolated string. The IDE doesn't care how you do it.

For example, if you create an interpolated string like this:
```C#
var value1 = $"The first value is { _portName }.";
var value2 = $"The second value is {_baudRate}.";
```
Notice that they are formatted differently. If you do a "Format Document", neither is changed.

**Additional information:**

This omission causes inconsistencies in codebases and should be rectified.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by jinujoseph** [comment]
Related to #31691

**Comment by CyrusNajmabadi** [comment]
Tracking through https://github.com/dotnet/roslyn/issues/31691
